### PR TITLE
tweak(rnd): Post infra change cleanup - fix process creation lifecycle

### PR DESCRIPTION
**Pull request by majdyz**
### Background

The websocket decoupling work is done, and there are some cleanup required in terms of the process execution and the way its life-cycle managed.

### Changes 🏗️

* Rename `poetry run app` --> `poetry run rest` and return the original meaning of app for running all app (excluding postgres & redis obviously) in a single command.
* Fix `wait_for_ns` hack on `run_processes` by introducing a `health_check` step.
* `rest` & `ws` now uses `run_processes` which properly kill the process on close & configure the logging.
* Disable freeze support & process configuration tweaking.

### PR Quality Scorecard ✨

- [x] Have you used the PR description template? &ensp; `+2 pts`
- [ ] Is your pull request atomic, focusing on a single change? &ensp; `+5 pts`
- [ ] Have you linked the GitHub issue(s) that this PR addresses? &ensp; `+5 pts`
- [ ] Have you documented your changes clearly and comprehensively? &ensp; `+5 pts`
- [ ] Have you changed or added a feature? &ensp; `-4 pts`
  - [ ] Have you added/updated corresponding documentation? &ensp; `+4 pts`
  - [ ] Have you added/updated corresponding integration tests? &ensp; `+5 pts`
- [ ] Have you changed the behavior of AutoGPT? &ensp; `-5 pts`
  - [ ] Have you also run `agbenchmark` to verify that these changes do not regress performance? &ensp; `+10 pts`

___

### **PR Type**
enhancement, bug_fix

___

### **Description**
- Refactored process execution and lifecycle management by removing `wait_for_nameserver` and integrating `health_check`.
- Added new entry points for REST and WebSocket API processes.
- Implemented `WebsocketServer` class for handling WebSocket API.
- Enhanced `PyroNameServer` with a health check functionality.
- Updated Dockerfile and `pyproject.toml` to reflect new process management changes.

___

### **Changes walkthrough** 📝
<table><thead><tr><th></th><th align="left">Relevant files</th></tr></thead><tbody><tr><td><strong>Enhancement</strong></td><td><details><summary>8 files</summary><table>
<tr>
  <td>
    <details>
      <summary><strong>app.py</strong><dd><code>Refactor process execution and lifecycle management</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

rnd/autogpt_server/autogpt_server/app.py

<li>Removed <code>wait_for_nameserver</code> and integrated <code>health_check</code>.<br> <li> Modified <code>run_processes</code> to handle processes more effectively.<br> <li> Added <code>WebsocketServer</code> to the list of processes.<br>

</details>

  </td>
  <td><a href="https://github.com/Significant-Gravitas/AutoGPT/pull/7981/files#diff-dc711625831f8d64dd1f36f49a8ab2ee02eadf6b01169c0b248f9617c40d4e18">+13/-28</a>&nbsp; </td>

</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>rest.py</strong><dd><code>Add entry point for REST API processes</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

rnd/autogpt_server/autogpt_server/rest.py

- Created a new entry point for running REST API processes.

</details>

  </td>
  <td><a href="https://github.com/Significant-Gravitas/AutoGPT/pull/7981/files#diff-e21e50d4fa02e8275c5c8c147a612f2aeb8c0b0054a58d90dc64f54430650597">+20/-0</a>&nbsp; &nbsp; </td>

</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>__init__.py</strong><dd><code>Export WebsocketServer in server module</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

rnd/autogpt_server/autogpt_server/server/__init__.py

- Added `WebsocketServer` to the module exports.

</details>

  </td>
  <td><a href="https://github.com/Significant-Gravitas/AutoGPT/pull/7981/files#diff-935005a6a22d405e844f6d8181799ef90f5c9a317c5ac61f208395cfff2f60a2">+2/-1</a>&nbsp; &nbsp; &nbsp; </td>

</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>ws_api.py</strong><dd><code>Implement WebsocketServer class for WebSocket API</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

rnd/autogpt_server/autogpt_server/server/ws_api.py

- Introduced `WebsocketServer` class for WebSocket handling.

</details>

  </td>
  <td><a href="https://github.com/Significant-Gravitas/AutoGPT/pull/7981/files#diff-1136b2eecfa149e4ae7a9ca2aee37d7270553dd09662d9da3c37b57d93bb8e75">+7/-0</a>&nbsp; &nbsp; &nbsp; </td>

</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>process.py</strong><dd><code>Add health check method to AppProcess</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

rnd/autogpt_server/autogpt_server/util/process.py

<li>Added <code>health_check</code> method to <code>AppProcess</code>.<br> <li> Integrated <code>health_check</code> into process start logic.<br>

</details>

  </td>
  <td><a href="https://github.com/Significant-Gravitas/AutoGPT/pull/7981/files#diff-3ca4e97f0542f3d39b3ef6d1df405d519bda8f463f6770618ed73dfee0a68d48">+7/-0</a>&nbsp; &nbsp; &nbsp; </td>

</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>service.py</strong><dd><code>Enhance PyroNameServer with health check functionality</code>&nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

rnd/autogpt_server/autogpt_server/util/service.py

- Implemented `_wait_for_ns` and `health_check` in `PyroNameServer`.

</details>

  </td>
  <td><a href="https://github.com/Significant-Gravitas/AutoGPT/pull/7981/files#diff-0688cfa91a411a53076603b85abc3541971cd2de1d2812b4f467ddfed1eea443">+8/-0</a>&nbsp; &nbsp; &nbsp; </td>

</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>ws.py</strong><dd><code>Add entry point for WebSocket API processes</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

rnd/autogpt_server/autogpt_server/ws.py

- Created a new entry point for running WebSocket API processes.

</details>

  </td>
  <td><a href="https://github.com/Significant-Gravitas/AutoGPT/pull/7981/files#diff-34a9fa4e06fe9465ad5fd47ceccceff95449999d76f50224d81f4c21467b9990">+13/-0</a>&nbsp; &nbsp; </td>

</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>ws_app.py</strong><dd><code>Remove standalone WebSocket application script</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

rnd/autogpt_server/autogpt_server/ws_app.py

- Removed the standalone WebSocket application script.

</details>

  </td>
  <td><a href="https://github.com/Significant-Gravitas/AutoGPT/pull/7981/files#diff-6f2a9695bf4fe8f895608d84f0eee001045a1b81c04f9396b3d110c5bb648552">+0/-11</a>&nbsp; &nbsp; </td>

</tr>                    
</table></details></td></tr><tr><td><strong>Configuration changes</strong></td><td><details><summary>2 files</summary><table>
<tr>
  <td>
    <details>
      <summary><strong>Dockerfile</strong><dd><code>Update Docker CMD to run REST API</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

rnd/autogpt_server/Dockerfile

- Updated CMD to run `rest` instead of `app`.

</details>

  </td>
  <td><a href="https://github.com/Significant-Gravitas/AutoGPT/pull/7981/files#diff-add6a932e871b7d444fbebe6037f796dadc17411c8189c27a45ea6cdc4f76e47">+1/-1</a>&nbsp; &nbsp; &nbsp; </td>

</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>pyproject.toml</strong><dd><code>Update script entry points in pyproject.toml</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

rnd/autogpt_server/pyproject.toml

- Renamed `app` script to `rest`.
- Updated `ws` script entry point.

</details>

  </td>
  <td><a href="https://github.com/Significant-Gravitas/AutoGPT/pull/7981/files#diff-adb694b1ef2fb9dabbdd0efe486d08bc9336bcd7b824f13f4125228d70cd4d7a">+2/-1</a>&nbsp; &nbsp; &nbsp; </td>

</tr>                    
</table></details></td></tr></tr></tbody></table>

___

> 💡 **PR-Agent usage**:
>Comment `/help` on the PR to get a list of all available PR-Agent tools and their descriptions

### CODE REVIEW

File: rnd/autogpt_server/autogpt_server/app.py [hunk 1]
```diff
@@ -1,61 +1,44 @@
...
-from tenacity import retry, stop_after_attempt, wait_exponential
-
 from .util.logging import configure_logging
 
 if TYPE_CHECKING:
     from autogpt_server.util.process import AppProcess
 
 
-@retry(stop=stop_after_attempt(30), wait=wait_exponential(multiplier=1, min=1, max=30))
-def wait_for_nameserver():
```
Comment (aarushik93) [inline rnd/autogpt_server/autogpt_server/app.py:14]: were you having an issue with this locally?

Comment (majdyz) [reply rnd/autogpt_server/autogpt_server/app.py:14]: I'm having a problem to call this websocket server only using this function, so I moved it on thr right place (def health check). The execution order will still be exactly the same.

Comment (aarushik93) [reply rnd/autogpt_server/autogpt_server/app.py:14]: > I'm having a problem to call this websocket server only using this function,

what do you mean? Can you clarify

Comment (majdyz) [reply rnd/autogpt_server/autogpt_server/app.py:14]: I can't do `run_processes(websocketserver())` or anything other than pyronameserver as the first entry.

This change is simply moving the `wait_for_ns` to block after calling the .start(background=True) on the. pyronameserver's part of the code, not here.

Comment (aarushik93) [reply rnd/autogpt_server/autogpt_server/app.py:14]: okay but i think we need to test this before merging as it's caused issues before

Comment (majdyz) [reply rnd/autogpt_server/autogpt_server/app.py:14]: how to test?

Comment (majdyz) [reply rnd/autogpt_server/autogpt_server/app.py:14]: Btw this doesn't change anything at all, it's just the placing of the code.

**Review by aarushik93 — APPROVED** [review]

**Review by ntindle — APPROVED** [review]

**Pull request merged**